arch/arm/core/thread.c

```diff
@@ -122,6 +122,11 @@ FUNC_NORETURN void _arch_user_mode_enter(k_thread_entry_t user_entry,
 	_current->arch.priv_stack_size =
 		(u32_t)CONFIG_PRIVILEGED_STACK_SIZE;
 
+#if defined (CONFIG_MPU_STACK_GUARD)
+	/* Re-program MPU to guard the privileged stack. */
+	configure_mpu_stack_guard(_current);
```

Comment:
A good catch.
But shouldn't we do this in userspace.S. To keep all MPU programing in one place rather than in multiple places.

Reply:
That's ok for me, I am going to update the patch.

Reply:
@AdithyaBaglody thanks for looking at it, btw.